source/atomics.tex

```diff
@@ -379,75 +379,90 @@
-\tcode{memory_order::seq_cst} fences \placeholder{X} and \placeholder{Y} such that \placeholder{A} is
-sequenced before \placeholder{X}, \placeholder{Y} is sequenced before \placeholder{B}, and \placeholder{X}
-precedes \placeholder{Y} in \placeholder{S}, then \placeholder{B} observes either the effects of
-\placeholder{A} or a later modification of \placeholder{M} in its modification order.
+\begin{note}
+This definition ensures that \placeholder{S} is consistent with
+the modification order of any atomic object \placeholder{M}.
+It also ensures that
+a \tcode{memory_order::seq_cst} load \placeholder{A} of \placeholder{M}
+gets its value either from the last modification of \placeholder{M}
+that precedes \placeholder{A} in \placeholder{S} or
+from some non-\tcode{memory_order::seq_cst} modification of \placeholder{M}
+that does not happen before any modification of \placeholder{M}
+that precedes \placeholder{A} in \placeholder{S}.
+\end{note}
 
 \pnum
-For atomic modifications \placeholder{A} and \placeholder{B} of an atomic object \placeholder{M},
-\placeholder{B} occurs later than \placeholder{A} in the modification order of \placeholder{M} if:
-
-\begin{itemize}
-\item there is a \tcode{memory_order::seq_cst} fence \placeholder{X} such that \placeholder{A}
-is sequenced before \placeholder{X}, and \placeholder{X} precedes \placeholder{B} in \placeholder{S}, or
-\item there is a \tcode{memory_order::seq_cst} fence \placeholder{Y} such that \placeholder{Y}
-is sequenced before \placeholder{B}, and \placeholder{A} precedes \placeholder{Y} in \placeholder{S}, or
-\item there are \tcode{memory_order::seq_cst} fences \placeholder{X} and \placeholder{Y} such that \placeholder{A}
-is sequenced before \placeholder{X}, \placeholder{Y} is sequenced before \placeholder{B},
-and \placeholder{X} precedes \placeholder{Y} in \placeholder{S}.
-\end{itemize}
-
+\begin{note}
+We do not require that \placeholder{S} be consistent with
+``happens before''\iref{intro.races}.
+This allows more efficient implementation
+of \tcode{memory_order::acquire} and \tcode{memory_order::release}
+on some machine architectures.
+It can produce surprising results
+when these are mixed with \tcode{memory_order::seq_cst} accesses.
+\end{note}
 
 \pnum
-\begin{note} \tcode{memory_order::seq_cst} ensures sequential consistency only for a
-program that is free of data races and uses exclusively \tcode{memory_order::seq_cst}
-operations. Any use of weaker ordering will invalidate this guarantee unless extreme
-care is used. In particular, \tcode{memory_order::seq_cst} fences ensure a total order
-only for the fences themselves. Fences cannot, in general, be used to restore sequential
-consistency for atomic operations with weaker ordering specifications. \end{note}
+\begin{note}
+\tcode{memory_order::seq_cst} ensures sequential consistency only
+for a program that is free of data races and
+uses exclusively \tcode{memory_order::seq_cst} operations.
```

Comment:
Missing the addition of "atomic" prior to "operations".

Reply:
Fixed.